modules/manager/controller_setup.go

```diff
@@ -354,6 +354,11 @@ func SetupControllers(mgr manager.Manager, c *Config, cpsMgr *multiinstance.Mana
 	if err := mgr.Add(scrapersMgr); err != nil {
 		return nil, fmt.Errorf("failed to add scrapers manager to controller-runtime manager: %w", err)
 	}
+	podLabels, err := k8sutils.GetSelfPodLabels()
+	if err != nil {
+		// TODO: output some log here?
+		podLabels = map[string]string{}
```

Comment:
I think that we could check if we're running in k8s (via a check on `KUBERNETES_SERVICE_HOST` env) and return an error then.

To allow running outside of the cluster I'd suggest to make sure failing to get self Pod labels is still supported (and does not create the network policy).

WDYT?

Reply:
Updated the logic like this:
- When running in k8s, return error if failed to get pod labels
- When running outside of k8s, bypass the check and do not create network policies
WDYT?